fvm/src/kernel/default.rs

```diff
@@ -136,10 +137,24 @@ where
             return Err(syscall_error!(LimitExceeded; "cannot store return block").into());
         }
 
+        self.call_manager
+            .get_actor_call_stack_mut()
+            .entry(self.actor_id)
+            .and_modify(|count| *count += 1)
+            .or_insert(1);
```

Comment:
This needs to happen within the call manager to handle the call into the top-level actor. It'll also help encapsulate things a bit.

I.e., I'd expose some form of `get_actor_recursion_count(actor_id)`, or something like that, but otherwise keep this datastructure internal (if possible).